plugins/database/src/mid_level_routines.cpp

```diff
@@ -1600,17 +1603,26 @@ int checkObjIdByTicket(const char* dataId,
 
     iUsesLimit = atoi( usesLimit );
     if ( iUsesLimit > 0 ) {
-        iUsesCount = atoi( usesCount );
-        if ( iUsesCount > iUsesLimit ) {
-            return CAT_TICKET_USES_EXCEEDED;
-        }
 
         intDataId = atoll( dataId );
 
         // Don't update a second time if this id matches the last one.
         // Given this function can be called multiple times within the same operation, checking to
         // see if the data id is different keeps the server from updating the ticket information multiple times.
         if ( previousDataId2 != intDataId ) {
+            errno = 0;
+            iUsesCount = static_cast<int>(strtol(usesCount, nullptr, 0));
```

Comment:
The easiest thing to do is replace use of strtol with stol and rely on exceptions. I bet those cover all the bases.

Reply:
I was just thinking about the case where you feed it garbage (e.g. "junk") and it fails to convert it to a value, effectively meaning that the uses count is 0.

I guess the choices for that case would be... 

a. the ticket is always usable
b. the ticket is never usable

This would fall under (a), which seems okay, I suppose. This only really matters if you can actually set the uses count, though. So maybe none of this matters (or at least is not particularly concerning).